### DISCUSSION TITLE
Mantine 7 styling solutions

### DISCUSSION THREAD

**Discussion by testerez** [mantinedev/mantine#5376] [Q&A]
Hello, I have a project using Mantine 6 and I'm investigating styling solutions to migrate to Mantine 7. Reading at the docs, my first impression is that any styling solution that would return CSS classenames would work well. But since the recommendation is to use either regular CSS or vanilla-extract, that makes me wonder how well Mantine would play with other styling solutions.

Here are a few candidate:
- [Emotion](https://emotion.sh/docs/introduction) (after all, can't I keep using emotion to generate classnames?)
- [Tailwind](https://tailwindcss.com/)
- [Panda](https://panda-css.com/)
- [StyleX](https://github.com/reacttips-dev/stylex)

Is there good reasons to stick to regular CSS or vanilla-extract, or are those valid alternatives?

**Comment by rtivital** [comment]
You can use any styling solution of your choice, there are some caveats, but overall all of them should work fine
https://mantine.dev/styles/css-modules/#styling-mantine-components-without-css-modules

**Reply by testerez** [reply]
Thanks @rtivital , the helpers in `@mantine/vanilla-extract` do not seam to be specific to vanilla-extract. Do you see any reason why I couldn't use them with [Panda](https://panda-css.com/) for example? And maybe it would make more sense to rename this package and update the Mantine doc to make it more obvious.

**Reply by THOUSAND-SKY** [reply]
Did you try with Panda? I tried but at least the `themeToVars` business isn't working out of the box. It generates classes like `text_var(--mantine-primary-color-filled)` which I don't think are valid for panda and don't work.

**Reply by testerez** [reply]
No I didn't yet
